vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -47,3 +58,119 @@ def preprocess(
             }
 
         return request
+
+    def preprocess_chat_request(
+        self, request: ChatDetectionRequest
+    ) -> Union[ChatDetectionRequest, ErrorResponse]:
+        """Granite guardian chat request preprocess is just detector parameter updates"""
+        return self.preprocess(request)
+
+    def request_to_chat_completion_request(
+        self, request: ContextAnalysisRequest, model_name: str
+    ) -> Union[ChatCompletionRequest, ErrorResponse]:
+        NO_RISK_NAME_MESSAGE = "No risk_name for context analysis"
+
+        risk_name = None
+        if (
+            "chat_template_kwargs" not in request.detector_params
+            or "guardian_config" not in request.detector_params["chat_template_kwargs"]
+        ):
+            return ErrorResponse(
+                message=NO_RISK_NAME_MESSAGE,
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+        # Use risk name to determine message format
+        if guardian_config := request.detector_params["chat_template_kwargs"][
+            "guardian_config"
+        ]:
+            risk_name = guardian_config["risk_name"]
+        else:
+            # Leaving off risk name can lead to model/template errors
+            return ErrorResponse(
+                message=NO_RISK_NAME_MESSAGE,
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+
+        if len(request.context) > 1:
```

Comment:
Can we concatenate contexts together if there are multiple and would that work upto certain context lengths ?